lib/rialto/etl/extractors/sera.rb

```diff
@@ -28,10 +29,16 @@ def each(&_block)
         attr_reader :sunetid
 
         def client
-          @client ||= ServiceClient::OauthConnectionFactory.build(service_url: ::Settings.sera.service_url,
-                                                                  client_id: ::Settings.sera.clientid,
-                                                                  client_secret: ::Settings.sera.secret,
-                                                                  token_url: ::Settings.sera.token_url)
+          oauth_client = OAuth2::Client.new(::Settings.sera.clientid,
```

Comment:
The way this is currently written, if I call client ten times, the first call will result in `@client` being set with the value of `oauth_client. The next nine calls (in this example) will reuse the `@client` value (because memoization) *but* it will still instantiate a new `OAuth2::Client` (and never use it). How about changing this method to look like:

```ruby
def client
  @client ||= begin
    oauth_client = OAuth2::Client.new(::Settings.sera.clientid,
                                      ::Settings.sera.secret,
                                      token_url: ::Settings.sera.token_url,
                                      auth_scheme: :request_body)

    ServiceClient::RetriableConnectionFactory.build(uri: ::Settings.sera.service_url,
                                                    headers: connection_headers,
                                                    oauth_token: oauth_client.client_credentials.get_token.token,
                                                    max_retries: ::Settings.sera.max_retries,
                                                    max_interval: ::Settings.sera.max_interval)
  end
end
```